crates/holochain/src/conductor/conductor/hc_p2p_handler_impl.rs

```diff
@@ -141,17 +141,33 @@ impl holochain_p2p::event::HcP2pHandler for Conductor {
         })
     }
 
-    fn handle_validation_receipts_received(
+    fn handle_validation_receipts(
         &self,
         dna_hash: DnaHash,
-        to_agent: AgentPubKey,
-        receipts: ValidationReceiptBundle,
-    ) -> BoxFut<'_, HolochainP2pResult<()>> {
-        Box::pin(async {
-            self.cell_by_parts(&dna_hash, &to_agent)
-                .await?
-                .handle_validation_receipts_received(dna_hash, to_agent, receipts)
-                .await
+        dht_op_list: Vec<DhtOpHash>,
+    ) -> BoxFut<'_, HolochainP2pResult<ValidationReceiptBundle>> {
+        Box::pin(async move {
+            let dht_db = self
+                .spaces
+                .dht_db(&dna_hash)
+                .map_err(HolochainP2pError::other)?;
+
+            let mut out = Vec::new();
+
+            for dht_op_hash in dht_op_list {
+                let mut receipts = dht_db
+                    .read_async(move |txn| {
+                        holochain_state::validation_receipts::list_receipts(txn, &dht_op_hash)
```

Comment:
Does the authority persist validation receipts? I thought they didn't. I had it in my head that validation receipts were sent immediately after they're generated.

Or rather, looking at `develop` - the validation receipt workflow builds the receipt bundles on demand for each op that is waiting on a receipt.

Reply:
Switched back to sign-on-demand: https://github.com/holochain/holochain/pull/4796/commits/f35e0b19653c20c6a65bf99d632435fe599b4379